### DISCUSSION TITLE
Meeting Minutes from March 7, 2024

### DISCUSSION THREAD

**Discussion by dezidizon** [CVWRSM/cvwrsm#67] [Meeting Minutes]
**Agenda:**
1. HecLib 7 has been tested for DCR, DCP, and LTO CS3 studies for v6 and v7 format.
2. DWR is working on Gradle downloadable JFreeChart and HDF5 libraries.
3. Timeline for merging Feature/wrims-devops branch into the master branch.
4. CalLite GUI Issue - [CalLiteGUI issue 220](https://github.com/CentralValleyModeling/CalLiteGUI/issues/220)

**Open Action Items:**
- @dwr-ycheng will coordinate a meeting with @rma-tom to address any question on the CalSim 3 model. This is tracked as #65.
- DWR to perform more comparison testing over a longer simulation period to verify that small differences do not increase over time. This is tracked as #66
- @rma-psmorris to coordinate a meeting with @inazrul and John DeGeorge.
- Coordinate with Hao Xie for manual steps in creating WRIMS installer and patch artifacts. This is tracked as #29.
- Apply copyright files (license and source) in the wrims repository using Eclipse Public License (EPL) version 2 or 3. This is tracked as #16.
- Remove Vista libraries from WRIMS. This is tracked as #33.

**Discussions:**
- @dwr-ycheng reported that @hxiedwr has tested the updates for HecLib 7 using DCR, DCP, and LTO CS3 studies producing DSS v6 and DSS v7 results.
    - @rma-tom requested a meeting to discuss the differences between the Delivery Capability Report (DCR), Delta Conveyance Project (DCP), and Long Term Operations (LTO) CalSim 3 models. (#65)
    - @hxiedwr reported that the DSS version 6 results from HecLib 6 and HecLib 7 have no difference. The DSS version 6 results from HecLib 6 and the DSS version 7 results from HecLib 7 have a small but acceptable difference.
    - @rma-tom recommended performing the comparison testing over a longer simulation to verify that small differences do not increase over time. (#66)
- @hxiedwr is working on having downloadable JFreeChart and HDF5 libraries to work with building WRIMS using Gradle. 
- @hxiedwr is interested in the timeline for merging the Feature/wrims-devops branch to the master branch in the wrims repository.
    -  The Feature/wrims-devops branch will be merged into the master branch once @rma-tom is done with work on the WRIMS plugin module.
- @dwr-ycheng discussed information needed to address the [issue with the CalLite GeoUI](https://github.com/CentralValleyModeling/CalLiteGUI/issues/220).
    - The issue was occurring in the  [CalLiteGUI](https://github.com/CentralValleyModeling/CalLiteGUI) repository. @dwr-ycheng will double-check if the issue is also occurring in the [CalLiteGUI Phase 3](https://github.com/CentralValleyModeling/CalLiteGUI_Phase3) repository. The fix will have to be submitted in both repositories.
    - @rma-psmorris proposed disabling the loading the maps when the XML does not exist. 
    - @dwr-psandhu noted that if the map service is not available, the schematic should still show. RMA will work with @dwr-ycheng on observing what the fix will look like when the fix has been implemented.
    - @inazrul noted that this is lower in priority compared to the work for WRIMS DevOps.
- @inazrul requested information regarding work progress and timeline for completing the WRIMS DevOps work.
    - @rma-psmorris and @dezidizon are actively working on preparing a report of work in progress and completed to date.
    - @inazrul noted that he is interested in moving to WRIMS 3.
    - @rma-psmorris will coordinate a meeting with @inazrul and John DeGeorge regarding funding to finish the WRIMS DevOps work.

**Closed Action Items:**
- @rma-tom to submit prototyping of comparison tests into Feature/wrims-devops. #54 
- @rma-tom to set-up artifact repository for hosting artifacts and watersheds required for automated testing. #53

**Comment by hxiedwr** [comment]
To me, it is not urgent to merge the master branch into WRIMS-DevOps for now. We still have some distance from releasing WRIMS 2 based on HecLib7 through DevOps auto build.
